group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3215,7 +3211,14 @@ public void replay(
         String groupId = key.groupId();
         String memberId = key.memberId();
 
-        ConsumerGroup consumerGroup = getOrMaybeCreatePersistedConsumerGroup(groupId, value != null);
+        ConsumerGroup consumerGroup;
+        try {
+            consumerGroup = getOrMaybeCreatePersistedConsumerGroup(groupId, value != null);
+        } catch (IllegalStateException ex) {
+            // If the group does not exist and a tombstone is replayed, we can ignore it.
```

Comment:
piggy backing for issue#1

To understand with an example: tombstone -> membership changes -> compaction: can have tombstones replay first - is this correct?

> if the __consumer_offsets partitions only contains tombstones for the group or the member.

Can a partition really only contain tombstones?

Reply:
> To understand with an example: tombstone -> membership changes -> compaction: can have tombstones replay first - is this correct?

Yes. The tombstone is replayed first in this case.

> Can a partition really only contain tombstones?

Yes. For instance, when all groups are deleted. However, in this case, we're talking about tombstones of one particular consumer group.

Reply:
Is there any other reason we could throw this IllegalStateException -- or just if the group is not found? I'm wondering if there is a clearer way to say this is a tombstone.

Reply:
Very good point! I think that we should rather use `GroupIdNotFoundException` for when the group does not exist and `IllegalStateException` when the state is not valid in order to differentiate the two cases. Let me fix.